python/pyspark/streaming/tests.py

```diff
@@ -1322,11 +1322,16 @@ def search_kinesis_asl_assembly_jar():
             "or 'build/mvn -Pkinesis-asl package' before running this test.")
 
     sys.stderr.write("Running tests: %s \n" % (str(testcases)))
+    failed = False
     for testcase in testcases:
         sys.stderr.write("[Running %s]\n" % (testcase))
         tests = unittest.TestLoader().loadTestsFromTestCase(testcase)
         if xmlrunner:
-            unittest.main(tests, verbosity=3,
```

Comment:
`unittest.main` will call sys.exit after finishing tests, so we should not use it here.